KEP 5075: update API description, add alternatives, update test plan

sunya-ch opened the pull request:
<!-- 
	Please use the following format when naming your PR
	< Issue Number >:< Issue Description >
	e.g. KEP-000: adding beta graduation criteria
	
	Avoid using phrases like `fixes #NNNN` in the description
	unless the pull request is to change the KEP status to 
	implemented or KEP has been deprecated.
-->

<!-- short description of work done in PR e.g. updating milestone, adding new KEP, adding test requirements… -->  
- One-line PR description: Update API descriptions, add alternatives and test plan those have been discussed and suggested after the [previous PR](https://github.com/kubernetes/enhancements/pull/5104/files) merged.

<!-- link to the k/enhancements issue -->
- Issue link: https://github.com/kubernetes/enhancements/issues/5075

<!-- other comments or additional information -->
- Other comments:

    What not included yet are:
    - Discussion about claimTracker
    -

johnbelamaric commented on keps/sig-scheduling/5075-dra-consumable-capacity/README.md:
```diff
@@ -122,11 +129,11 @@ and that resource claim has allocated the device.
 
-Additionally, if a device supports sharing, its resource (capacity) can be managed through a defined sharing policy.
+Additionally, if a device supports sharing, its resource (capacity) can be managed through a defined request policy.
 When such a policy is specified, the device may be allocated to multiple independent requests, up to its total capacity,
```
this needs to be re-written now that allow multiple allocations and request policy are decoupled.

johnbelamaric reviewed (commented):
Started reviewing but I think it needs some rework for the most recent changes, so I stopped pretty early. Once all the changes from the implementation phase are incorporated, I will take another look.

liggitt commented:
+1 to folding in updates to the field names and behavior made during implementation review

sunya-ch commented:
@liggitt @johnbelamaric I updated the description and API according to the recent API in the implementation.
I also removed the section `Device driver migration` since we do not need the driver to prevent unintentionally sharing when we set default to full capacity.

pohly commented on keps/sig-scheduling/5075-dra-consumable-capacity/README.md:
```diff
@@ -179,338 +187,396 @@ In other words, the device capacity allocation is determined by the user's claim
+	//
+	// +optional
+	// +featureGate=DRAConsumableCapacity
+	AllowMultipleAllocations *bool
```
I prefer to not use tabs in the README.md. Better replace with four spaces.

SergeyKanzhelev commented:
I believe this needs to be merged: https://github.com/kubernetes/enhancements/pull/5634/ at a minimum if we plan to make changes to this feature

macsko commented on keps/sig-scheduling/5075-dra-consumable-capacity/README.md:
```diff
@@ -101,7 +107,7 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
   - [ ] e2e Tests for all Beta API Operations (endpoints)
   - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
-  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+  - [ ] (R) Min Two Week Window for GA e2e tests to prove flake free
```
nit, revert:
```suggestion
  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
```

dom4ha commented on keps/sig-scheduling/5075-dra-consumable-capacity/README.md:
```diff
@@ -179,338 +189,398 @@ In other words, the device capacity allocation is determined by the user's claim
 
-If users do not specify a capacity request for consumable capacity, the default consumed value will be applied.
-There is always such a default because capacities without a policy are not consumable.
+If users do not specify a capacity request, the default consumed value will be the device’s full capacity or the default value specified by the request policy.
```
```suggestion
If users do not specify a capacity request, the consumed value will be:
- the device’s full capacity or
- the default value if it was specified by the request policy or
- none if the device usage is unlimited
```
